### PR TITLE
No sig slasher rpc

### PR DESCRIPTION
**Pull request by shayzluf**
**What type of PR is this?**

> Uncomment one line below and remove others.
Feature

**What does this PR do? Why is it needed?**
adds two endpoints to slasher grpc server that detect slashable blocks and attestations and doesn't update the slasher db.

those endpoints are needed to protect validators from signing blocks and attestations that could be later be found to be slashable.

**Which issues(s) does this PR fix?**

Part of #5921

### CODE REVIEW

File: slasher/detection/proposals/detector.go [hunk 1]
```diff
@@ -51,3 +51,26 @@ func (dd *ProposeDetector) DetectDoublePropose(
...
+	headersFromIdx, err := dd.slasherDB.BlockHeaders(ctx, incomingBlk.Slot, incomingBlk.ProposerIndex)
+	if err != nil {
+		return false, err
+	}
+	for _, blockHeader := range headersFromIdx {
+		sameBodyRoot := bytes.Equal(blockHeader.Header.BodyRoot, incomingBlk.BodyRoot)
+		sameSatateRoot := bytes.Equal(blockHeader.Header.StateRoot, incomingBlk.StateRoot)
+		sameParentRoot := bytes.Equal(blockHeader.Header.ParentRoot, incomingBlk.ParentRoot)
+		if sameBodyRoot && sameSatateRoot && sameParentRoot {
+			continue
```
Comment (prestonvanloon) [inline slasher/detection/proposals/detector.go:71]:
```suggestion
		sameBodyRoot := bytes.Equal(blockHeader.Header.BodyRoot, incomingBlk.BodyRoot)
		sameStateRoot := bytes.Equal(blockHeader.Header.StateRoot, incomingBlk.StateRoot)
		sameParentRoot := bytes.Equal(blockHeader.Header.ParentRoot, incomingBlk.ParentRoot)
		if sameBodyRoot && sameStateRoot && sameParentRoot {
			continue
```

File: slasher/detection/proposals/detector.go [hunk 2]
```diff
@@ -51,3 +51,26 @@ func (dd *ProposeDetector) DetectDoublePropose(
 	}
 	return nil, nil
 }
+
+// DetectDoubleProposeNoUpdate detects double proposals for a given block header by looking in the db.
```
Comment (prestonvanloon) [inline slasher/detection/proposals/detector.go:55]: What does "NoUpdate" mean?

Comment (shayzluf) [reply slasher/detection/proposals/detector.go:55]: Without db update that will result in effecting other proposals detection

File: slasher/detection/proposals/detector.go (hunk 1 above)
Comment (shayzluf) [reply slasher/detection/proposals/detector.go:71]: thanks

File: slasher/detection/proposals/detector.go (hunk 2 above)
Comment (shayzluf) [reply slasher/detection/proposals/detector.go]: updated the comment . hope its ok now

File: slasher/rpc/server.go [hunk 3]
```diff
@@ -104,7 +104,7 @@ func (ss *Server) IsSlashableAttestation(ctx context.Context, req *ethpb.Indexed
 }
 
 // IsSlashableBlock returns an proposer slashing if the block submitted
-// is a double proposal.
+// is a double proposal(no db update is being done).
```
Comment (0xKiwi) [inline slasher/rpc/server.go:107]:
```suggestion
// is a double proposal (no db update is being done).
```

Comment (0xKiwi) [reply slasher/rpc/server.go:107]: Also, this comment could be misleading, `DetectDoubleProposals` saves all blocks passed into it, is does update the DB.

File: slasher/rpc/server.go [hunk 4]
```diff
@@ -150,3 +150,30 @@ func (ss *Server) IsSlashableBlock(ctx context.Context, req *ethpb.SignedBeaconB
 	return psr, nil
 
 }
+
+// IsSlashableAttestationNoUpdate returns true if the attestation submitted
+// is a slashable vote(no db update is being done).
+func (ss *Server) IsSlashableAttestationNoUpdate(ctx context.Context, req *ethpb.IndexedAttestation) (*slashpb.Slashable, error) {
+	slashings, err := ss.detector.DetectAttesterSlashings(ctx, req)
+	sl := &slashpb.Slashable{Slashable: false}
```
Comment (0xKiwi) [inline slasher/rpc/server.go:158]: Do you think you could move this higher up? Also no need to declare it if its going to be false `Slashable{}` would default to false, so you could just return `Slashable{Slashable: true}` and the end.

File: slasher/rpc/server.go [hunk 5]
```diff
@@ -150,3 +150,30 @@ func (ss *Server) IsSlashableBlock(ctx context.Context, req *ethpb.SignedBeaconB
...
+	return sl, nil
+}
+
+// IsSlashableBlockNoUpdate returns true if the block submitted
+// is slashable.
+func (ss *Server) IsSlashableBlockNoUpdate(ctx context.Context, req *ethpb.BeaconBlockHeader) (*slashpb.Slashable, error) {
+	slash, err := ss.detector.DetectDoubleProposeNoUpdate(ctx, req)
+	if err != nil {
+		sl := &slashpb.Slashable{Slashable: false}
+		return sl, status.Errorf(codes.Internal, "Could not detect proposer slashing for block: %v: %v", req, err)
```
[the remark this answers is not in the document]
Comment (shayzluf) [reply slasher/rpc/server.go:175]: thanks

File: slasher/rpc/server.go [hunk 6]
```diff
@@ -150,3 +150,30 @@ func (ss *Server) IsSlashableBlock(ctx context.Context, req *ethpb.SignedBeaconB
...
 
 }
+
+// IsSlashableAttestationNoUpdate returns true if the attestation submitted
+// is a slashable vote(no db update is being done).
+func (ss *Server) IsSlashableAttestationNoUpdate(ctx context.Context, req *ethpb.IndexedAttestation) (*slashpb.Slashable, error) {
+	slashings, err := ss.detector.DetectAttesterSlashings(ctx, req)
+	sl := &slashpb.Slashable{Slashable: false}
+	if err != nil {
+		return sl, status.Errorf(codes.Internal, "Could not detect attester slashings for attestation: %v: %v", req, err)
```
[the remark this answers is not in the document]
Comment (shayzluf) [reply slasher/rpc/server.go:160]: thanks

File: slasher/rpc/server.go (hunk 3 above)
Comment (shayzluf) [reply slasher/rpc/server.go:107]: good catch . the right comment for the wrong function

File: slasher/rpc/server.go (hunk 4 above)
Comment (shayzluf) [reply slasher/rpc/server.go]: changed. thanks

File: slasher/rpc/server.go (hunk 3 above)
Comment (shayzluf) [reply slasher/rpc/server.go]: fixed

File: proto/slashing/slashing.proto [hunk 7]
```diff
@@ -17,12 +17,24 @@ service Slasher {
 
     // Returns any found proposer slashings if the passed in proposal conflicts with a validators history.
     rpc IsSlashableBlock(ethereum.eth.v1alpha1.SignedBeaconBlockHeader) returns (ProposerSlashingResponse);
+
+    // Returns any found attester slashings if the passed in attestation conflicts with a validators history.
+    // doesn't preform db update(signature is not required).
+    rpc IsSlashableAttestationNoUpdate(ethereum.eth.v1alpha1.IndexedAttestation) returns (Slashable);
+
+    // Returns any found proposer slashings if the passed in proposal conflicts with a validators history.
+    // doesn't preform db update(signature is not required).
```
Comment (0xKiwi) [inline proto/slashing/slashing.proto:26]:
```suggestion
    // Returns if a given beacon block header could be slashable when compared to the slashers history for the proposer.
    // This function is read-only, and does not need the beacon block header to be signed.
```

File: proto/slashing/slashing.proto [hunk 8]
```diff
@@ -17,12 +17,24 @@ service Slasher {
 
     // Returns any found proposer slashings if the passed in proposal conflicts with a validators history.
     rpc IsSlashableBlock(ethereum.eth.v1alpha1.SignedBeaconBlockHeader) returns (ProposerSlashingResponse);
+
+    // Returns any found attester slashings if the passed in attestation conflicts with a validators history.
+    // doesn't preform db update(signature is not required).
```
Comment (0xKiwi) [inline proto/slashing/slashing.proto:22]:
```suggestion
    // Returns if a given indexed attestation could be slashable when compared to the slashers history for the attesters.
    // This function is read-only, and does not need the indexed attestation to be signed.
```

**Pull request merged**